ui.tests/test-module/specs/recaptcha/recaptcha.runtime.spec.js

```diff
@@ -0,0 +1,98 @@
+    const bemBlock = 'cmp-adaptiveform-recaptcha'
+    const IS = "adaptiveFormRecaptcha"
+    const selectors = {
+        recaptcha : `[data-cmp-is="${IS}"]`
+    }
+    let toggle_array = []
+
+    let formContainer = null
+
+    beforeEach(() => {
+        cy.fetchFeatureToggles().then((response) => {
+            if (response.status === 200) {
+                toggle_array = response.body.enabled;
+                if (toggle_array && toggle_array.includes(FT_CLOUD_CONFIG_PROVIDER)) {
+                    cy.previewForm(pagePath).then((p) => {
+                        formContainer = p;
+                    });
+                }
+            }
+        });
+    });
+
+    const checkHTML = (id, state) => {
+        const visible = state.visible;
+        const passVisibleCheck = `${visible === true ? "" : "not."}be.visible`;
+        const passDisabledAttributeCheck = `${state.enabled === false || state.readOnly === true ? "" : "not."}have.attr`;
+        const value = state.value
+        cy.get(`#${id}`)
+            .should(passVisibleCheck)
+            .invoke('attr', 'data-cmp-visible')
+            .should('eq', visible.toString());
+        cy.get(`#${id}`)
+            .invoke('attr', 'data-cmp-enabled')
+            .should('eq', state.enabled.toString());
+        return cy.get(`#${id}`).within((root) => {
+            cy.get('*').should(passVisibleCheck)
+        })
+    }
+
+    it(" should get model and view initialized properly ", () => {
+        if(toggle_array && toggle_array.includes(FT_CLOUD_CONFIG_PROVIDER)) {
+            expect(formContainer, "formcontainer is initialized").to.not.be.null;
+            expect(formContainer._model.items.length, "model and view elements match").to.equal(Object.keys(formContainer._fields).length);
+            Object.entries(formContainer._fields).forEach(([id, field]) => {
+                expect(field.getId()).to.equal(id)
+                expect(formContainer._model.getElement(id), `model and view are in sync`).to.equal(field.getModel())
+            });
+        }
+    })
+
+    it(" model's changes are reflected in the html ", () => {
+        if(toggle_array && toggle_array.includes(FT_CLOUD_CONFIG_PROVIDER)) {
+            const [id, fieldView] = Object.entries(formContainer._fields)[0]
+            const model = formContainer._model.getElement(id)
+            cy.get('#' + id + ' .cmp-adaptiveform-recaptcha__widget > div.g-recaptcha').should('exist');
+
+            checkHTML(model.id, model.getState()).then(() => {
+                model.visible = false
+                return checkHTML(model.id, model.getState())
+            }).then(() => {
+                model.enable = false
+                return checkHTML(model.id, model.getState())
+            })
+        }
+    });
+
+    it(" html changes are reflected in model ", () => {
+        if(toggle_array && toggle_array.includes(FT_CLOUD_CONFIG_PROVIDER)) {
```

Comment:
Please enable feature toggle by uploading the SDK, why are you check if present in toggle array

Reply:
Something like this, https://github.com/adobe/aem-core-forms-components/blob/dev/it/config/src/main/content/jcr_root/apps/system/config/com.adobe.granite.toggle.impl.dev.DynamicToggleProviderImpl.cfg.json